src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java

```diff
@@ -217,4 +216,17 @@ String computeCrc32c(String input) {
         return outputs;
     }
 
+    private <V> boolean shouldRetry(Throwable throwable, V input) {
```

Comment:
I don't think it's part of this commit but could this get stuck in a loop?  e.g. is there an upper limit to the number of retries?

Reply:
The batching code has a check: if it doesn't make any progress after `NO_PROGRESS_MAX` retries, it gives up. That is currently set to 2.

Reply:
Got it, thanks!